docs/rst/manual/_ext/ert_jobs.py

```diff
@@ -0,0 +1,215 @@
+import sys
+
+import docutils.nodes
+import docutils.parsers.rst
+import docutils.statemachine
+from docutils import nodes
+from ert_shared.plugins import ErtPluginManager
+from sphinx.util.docutils import SphinxDirective
+from sphinx.util.nodes import nested_parse_with_titles
+
+
+def _parse_raw_rst(rst_string, node, state):
+    string_list = docutils.statemachine.StringList(list(rst_string.split("\n")))
+    _parse_string_list(string_list, node, state)
+
+
+def _parse_string_list(string_list, node, state):
+    nested_parse_with_titles(state, string_list, node)
+
+
+def _escape_id(id_string):
+    return id_string.replace(" ", "-")
+
+
+def _create_section_with_title(section_id, title):
+    node = nodes.section(ids=[_escape_id(section_id)])
+    jobs_title = nodes.title(text=title)
+    node.append(jobs_title)
+    return node
+
+
+class ForwardModelDocumentation:
+    def __init__(
+        self, name, category, job_source, description, job_config_file, examples="",
+    ):
+        self.name = name
+        self.job_source = job_source
+        self.category = category
+        self.description = description
+        self.job_config_file = job_config_file
+        self.examples = examples
+
+    def _create_job_config_section(self):
+        config_section_node = _create_section_with_title(
+            section_id=self.name + "-job-config", title="ERT config file"
+        )
+
+        with open(self.job_config_file) as fh:
+            job_config_text = fh.read()
+        job_config_text_node = nodes.literal_block(text=job_config_text)
+        config_section_node.append(job_config_text_node)
+        return config_section_node
+
+    def _create_job_details(self):
+        job_category_node = nodes.definition_list_item(
+            "",
+            nodes.term("", "", nodes.strong(text="Category")),
+            nodes.definition("", nodes.strong(text=self.category)),
+        )
+        job_package_node = nodes.definition_list_item(
+            "",
+            nodes.term("", "", nodes.strong(text="Source package")),
+            nodes.definition("", nodes.strong(text=self.job_source)),
+        )
+        job_details_node = nodes.definition_list(
+            "", job_category_node, job_package_node
+        )
+        return job_details_node
+
+    def _create_example_section(self, state):
+        example_section_node = _create_section_with_title(
+            section_id=self.name + "-examples", title="Examples"
+        )
+        _parse_raw_rst(self.examples, example_section_node, state)
+        return example_section_node
+
+    def create_node(self, state):
+        # Create main node section
```

Comment:
this is just a comment - not sure if it makes sense - but maybe somthing to reflect upon. 
would it be even more clean ( this code looks really good! ) if the comments was replaced by some actual private funcs like `def _create_main_node_section(..)` , `def _add_description_and_fm_details(..)` .... ? 

when stumbling upon this code 3 years down the road i guess new lines are added underneath some of the sections making the comments a lie.... 

but - that was just  a reflection 😺

Reply:
I agree, and at some point I actually had it like that. However it ended up with very many functions, to the point where I questioned the readability of the code because it ended being a lot of 2 line functions. So I settled on this, where the the content nodes are created in functions and then appended to the main node, separated with comments. I think that strikes a balance between readability and avoiding confusing mutating of the main node objects in all of the functions :slightly_smiling_face:

Reply:
totally agree... the important thing is you reflected upon it 👍  it should make sense when refactoring and splitting everything out in functions - sometimes it's just getting things worse